core/src/main/scala/scalaz/Foldable.scala

```diff
@@ -325,6 +325,9 @@ trait Foldable[F[_]]  { self =>
         else seen
     }.reverse
 
+  def distinctBy[A, B: Equal](fa: F[A])(f: A => B): IList[A] =
+    distinctE(fa)(Equal.equalBy(f))
```

Comment:
@jbgi Perhaps you want the type of `seen` to be `IList[B]` to avoid allocation of two `Function1`s per each element of `fa`.